authors/apps/authentication/views.py

```diff
@@ -72,3 +77,59 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class PasswordResetAPIView(CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = PasswordResetSerializer
+
+    def post(self, request):
+        recipient = request.data.get('email', {})  # user enters email
+        if not recipient:
+            return Response({"message": "Please provide an email address"},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        # generate a new token for email entered
+        token = jwt.encode({"email": recipient},
+                           SECRET_KEY, algorithm='HS256')
+        # confirm if user exists
+        user_exists = User.objects.filter(email=recipient).exists()
+        if user_exists:
+            result = send_email(recipient, token, request)
+            return Response(result, status=status.HTTP_200_OK)
+        else:  # if user does not exist
+            result = {
+                'message': 'Sorry, user with this email does not exist'
+            }
+            return Response(result, status=status.HTTP_404_NOT_FOUND)
+
+
+class PasswordUpdateAPIView(UpdateAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = PasswordResetSerializer
+
+    def update(self, request, *args, **kwargs):
+        token = self.kwargs.get('token')
+        password = request.data.get('password')
+        confirm_password = request.data.get('confirm_password')
+        if password != confirm_password:
+            return Response({"message": "Sorry, your passwords do not match"},
+                            status=status.HTTP_400_BAD_REQUEST)
+        password = {
+            "password": password
+        }
+        serializer = self.serializer_class(data=password)
+        serializer.is_valid(raise_exception=True)
+
+        try:
+            # decode token and save new password
+            decode_token = jwt.decode(
+                token, SECRET_KEY, algorithms=['HS256'])
+            user = User.objects.get(email=decode_token['email'])
+            user.set_password(password)
+            user.save()
```

Comment:
The password you are setting is an object above instead of the real password provided by the user.

Reply:
Hey, @sam-karis I've looked into that and you were right. I have fixed the issue and it's now setting the password as a string. I realized I used the same variable names(password) so I just had to do a ```user.set_password(request.data.get('password'))```. Please check to review the changes.